gandi_tf/main.py

```diff
@@ -87,33 +123,54 @@ def generate_tf(domain, entries):
         f.write("  type   = each.value.type\n")
         f.write("  values = each.value.values\n")
         f.write("}\n")
-
+    if subdir is True:
+        with click.open_file(filename_tfimport, "w") as fi:
+            for cmd in commands:
+                fi.write(f"{cmd}\n")
     return commands
 
 
 @click.command(no_args_is_help=True)
 @click.option("--version", help="Display version information", is_flag=True)
+@click.option("--organization_id", help="To list domains only owned by this organization id", default=None)
+@click.option('--nsfilters', type=click.Choice(['abc','livedns','other'], case_sensitive=True), help="Filter domaine based on their current nameserver 'abc','livedns' or 'other'. You can add multiple filter Example: gandi2tf --list_domains --nsfilters abc --nsfilters livedns)", multiple=True, default=['abc','livedns','other'])
+@click.option("--auto-generate", help="Auto-generate tf resources based on domain from gandi instead of STDIN/USERINPUT (option can be used with --organization_id, --subdir and multiple --nsfilters)", is_flag=True)
+@click.option("--subdir", help="Create a sub-directory to store generated domain tf and tf import command", is_flag=True)
 @click.argument("domains", nargs=-1)
-def generate(domains, version):
+def generate(domains, version, organization_id, nsfilters, auto_generate, subdir):
     """
     Command to read Gandi.net live DNS records and generate
     corresponding TF gandi_livedns_record resources.
+
+    Warning: nameserver type 'other' and 'abc' can't be managed via terraform (for abc domain you can transfer them to livedns on gandi webinterface)
```

Comment:
Due to my mistake I have to ask question afterwards... Why did you add the option `nsfilters` to contain the non-compatible values?

Reply:
Hi @marcaurele 

Well nsfilters was design to work with the auto-generate feature, for example i'm using it to retrieve and generate tf and import state file of all domain available in my organisation that are currently manage via gandi livedns filtering out automatically  domain we own currently delegated to other provider   

Example of command fetching all domain managed by organization XYZ and placing them in subdir:
gandi2tf  --organization_id XYZ --autogenerate --nsfilters livedns --subdir 

After review of the element you chose to merge into your project you can remove reference to nsfilters, organization_id, and fetch_domains_list function and call to function 

If i'm not mistaken you will endup with feature --subdir and the patch for domain starting with number

If you need me to do a cleanup and propose a new version via MR let me know

Sincerely
Dopch

Reply:
i only have entries with livedns. My understanding of your explanation is that it should only retrieve the _livedns_ entries, the other one should be left out of the TF code. Therefore it should force the `nameserver` paramter to `livedns`.

Reply:
See https://github.com/marcaurele/gandi-2-terraform/pull/11/commits/d1ff185d2dc69ce0d8134147f6eeb0410a0ffa6d

Reply:
fetch with the filter livedns is indeed correct for your project scope